bokehjs/src/lib/core/util/color.ts

```diff
@@ -64,7 +77,12 @@ export function color2rgba(color: string | null, alpha: number = 1.0): RGBA {
     rgba.push(0)
   if (rgba.length < 4)
     rgba.push(alpha)
-  return rgba.slice(0, 4) as RGBA
+
+  const rgbaf = rgba.slice(0, 4) as RGBAf
+  _last_color = color
+  _last_alpha = alpha
+  _last_rgbaf = rgbaf
```

Comment:
Is there evidence that we are repeatedly calling `color2rgba` on sequences that contain runs of the same color?

Reply:
Yes, especially until 2.3 performance regressions are addressed, but ultimately there should be a cache of most often used colors, as parsing CSS color strings pretty expensive. Though this particular implementation can be improved and I already did that in my other work (where I add support for CSS4 color strings; a PR will come after this one is merged).